wgpu-hal/src/gles/wgl.rs

```diff
@@ -101,7 +108,10 @@ impl<'a> std::ops::Deref for AdapterContextLock<'a> {
 
 impl<'a> Drop for AdapterContextLock<'a> {
     fn drop(&mut self) {
-        self.inner.context.unmake_current().unwrap();
+        // EGL performs a .take() here but that causes issues with drop ordering
+        if let Some(wgl) = &self.inner.context {
```

Comment:
I had a `.take()` here that may be the cause of all the spurious CI failures at https://github.com/gfx-rs/wgpu/actions/runs/10584695195/job/29329702561.

EGL does it too though, and unmaking the context current here pretty much defeats the purpose of #6114 doesn't it?

Reply:
The patterns used for this lock guard between EGL and WGL aren't really consistent. The `.take()` in EGL isn't taking from the `egl` field in `AdapterContext` (which would have a similar effect to doing `self.inner.context.take()` here), instead it is a field in a different type that just holds a reference and I think a copy of the Display pointer (tbh it doesn't seem like the take is not actually necessary in EGL since the used values are `Copy`).

Reply:
Right, yes. Perhaps we should create a `WglContextLock` too and/or map the lock to `glow::Context` instead of lockign all of `Inner`?

Reply:
> Right, yes. Perhaps we should create a `WglContextLock` too and/or map the lock to `glow::Context` instead of lockign all of `Inner`?
>

Reply:
Perhaps yeah. An alternative is moving the EGL implementation to closer match the WGL one. I don't know if there are any differences between EGL and WGL that pushed them to these different approaches though.

> unmaking the context current here pretty much defeats the purpose of https://github.com/gfx-rs/wgpu/pull/6114 doesn't it?

Is this question still relevant or is it resolved now that the `take` behavior is clarified?

Reply:
Making WGL look more like EGL is mostly an annoying refactor, but I don't see any reason why it cannot be done or why it wasn't like this in the first place.

Reply:
@MarijnS95: Is this matter resolved? I wish I understood it well enough to tell immediately.

Reply:
@ErichDonGubler it's working as-is but the two backend implementations are unnecessarily hard to compare.

Reply:
In other words, I'm not sure if I should strive to bring any consistency/improvement here (mainly: seeing why different parts are behind or outside the `Mutex<>` in either backend) but that could/should strictly go in a separate followup PR.

Reply:
If it really is just a refactor to make things "nice" outstanding, I'm inclined to review and (when we get to that point) merge, then free up resources for the refactor in everybody's brains later.